s3storageprovider/src/test/java/org/duracloud/s3storage/S3ProviderUtilTest.java

```diff
@@ -32,4 +38,25 @@ public void testGetBucketName() throws Exception {
                                                         "x--..y..--z-.");
         assertEquals("abc.x-y.z", bucketName);
     }
+
+    @Test
+    public void testGetAmazonS3Client() {
+        String accessKey = "access-key";
+        String privateKey = "private-key";
+
+        Map<String, String> optionsA = new HashMap<>();
+        optionsA.put(AWS_REGION.name(), "us-east-1");
+
+        Map<String, String> optionsB = new HashMap<>();
+        optionsB.put(AWS_REGION.name(), "us-west-2");
+
+        AmazonS3 s3ClientA = S3ProviderUtil.getAmazonS3Client(accessKey, privateKey, optionsA);
+        assertEquals("us-east-1", s3ClientA.getRegionName());
+
+        AmazonS3 s3ClientB = S3ProviderUtil.getAmazonS3Client(accessKey, privateKey, optionsB);
+        assertEquals("us-west-2", s3ClientB.getRegionName());
+
+        assertNotSame(s3ClientA, s3ClientB);
+    }
```

Comment:
What do you think about adding  a second test that demonstrates that two clients built with the same logical options (but using separate hashmap instance) returns the same s3 client instance?